### PR TITLE
adding docs generation

### PR DESCRIPTION
**Pull request by fschlimb**
Adding generation of documentation, following procedures from upstream MLIR.
A new directory `docs` provides manually created top-level `.md` docs wich refer to others, including Docs for passes and dialects, which are auto-generated by tablegen and copied to the docs folder appropriately.

Please review these guidelines to help with the review process:
- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, a reproducer, or a reference to an issue with a reproducer?
- [ ] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?
- [x] Have you organized your commits logically and ensured each can be built by itself?

### CODE REVIEW

**Comment by fschlimb** [comment]
@Hardcode84 @nbpatel @drprajap Anyone?

File: include/imex/Transforms/Passes.td [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+//
+// Copyright 2022 Intel Corporation
+// Part of the IMEX Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file defines the base classes of IMEX conversino passes.
```
Comment (drprajap) [inline include/imex/Transforms/Passes.td:11]: nit: typo
```suggestion
/// This file defines the base classes of IMEX conversion passes.
```

**Review by drprajap — APPROVED** [review]

**Comment by drprajap** [comment]
I quickly checked the build and there are bunch of error and warning about dot, should we install graphviz separately to address that or not worrying about it for now?

File: include/imex/Transforms/Passes.td (hunk 1 above)
Comment (fschlimb) [reply include/imex/Transforms/Passes.td]: Fixed

**Comment by fschlimb** [comment]
> I quickly checked the build and there are bunch of error and warning about dot, should we install graphviz separately to address that or not worrying about it for now?

Fixed, I also eliminated a few other warnings (including adding a mainpage).

**Comment by fschlimb** [comment]
last call.

**Review by diptorupd — APPROVED** [review]

**Comment by nbpatel** [comment]
I just had a comment about whether we should use generation by building or just add the .md files or both...upstream has some .md files in the docs directory. I had a short discussion with Frank about this.any comments from @drprajap or @diptorupd ?

**Comment by fschlimb** [comment]
> I just had a comment about whether we should use doc generation by building or just add the .md files in the doc directory or both...upstream has some .md files in the docs directory which dont seem to be generated. I had a short discussion with Frank about this.any comments from @drprajap or @diptorupd ?

I updated the README with brief description of the standard procedure to document dialects and conversions. Having the doc within the .td files puts the source code and the doc together in the same place and automatically generates a standardized document structure across all dialects/conversions.

**Comment by drprajap** [comment]
LGTM. when if future if we plan to add Spinix documentation for API/interface, we can render this in separate github website, but at this moment, this is good to merge.

**Pull request merged**